ephios/plugins/eventautoqualification/consequences.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import operator
+from collections import Counter
+
+from django.db.models import Q
+
+from ephios.core.consequences import QualificationConsequenceHandler
+from ephios.core.models import AbstractParticipation, LocalParticipation
+from ephios.plugins.eventautoqualification.models import EventAutoQualificationConfiguration
+
+logger = logging.getLogger(__name__)
+
+
+def create_qualification_consequence(sender, participation, **kwargs):
+    event = participation.shift.event
+    try:
+        mode = event.auto_qualification_config.mode
+    except EventAutoQualificationConfiguration.DoesNotExist:
+        return
+
+    # check if the required shift participations exist
+    requirements_met = False
+    if mode == EventAutoQualificationConfiguration.Modes.ANY_SHIFT:
+        requirements_met = True
+    elif mode == EventAutoQualificationConfiguration.Modes.LAST_SHIFT:
+        requirements_met = (
+            participation.shift
+            == sorted(event.shifts.all(), key=operator.attrgetter("end_time"))[-1]
```

Comment:
why sort in python and not on the queryset?

Reply:
To not cause another dB hit